src/callbacks/callback_helpers.jl

```diff
@@ -59,18 +59,16 @@ callback_from_affect(x::AtmosCallback) = x
 function callback_from_affect(affect!)
     for p in propertynames(affect!)
         x = getproperty(affect!, p)
-        if x isa AtmosCallback
-            return x
-        elseif x isa DECB.SavedValues
+        if x isa AtmosCallback || x isa DECB.SavedValues
             return x
         end
     end
-    error("Callback not found in $(affect!)")
+    return nothing
```

Comment:
Should we remove this error? It might otherwise lead to confusing errors?

Reply:
Good question.

The function forced all the callbacks to be either `AtmosCallbacks` or a specific type of `DiffEqCallbacks` callback (one with the `SavedValues` property), then the function `atmos_callbacks` filtered different types of callbacks dropping the `DiffEqCallbacks` callbacks. I flipped the conditional around so that `atmos_callbacks` filters for `AtmosCallbacks` (which is really the intent here: we want `AtmosCallbacks` so that downstream functions can compute lcm and cycle). Therefore, there is no longer a reason to reject callbacks that are not `AtmosCallbacks` or `DiffEqCallbacks`. 

The diagnostic callback is a `SciMLBase.DiscreteCallback`, hence, it was hitting the error banch here. Note that there is no reason for the diagnostic callback to be an AtmosCallback as it doesn't have a strong notion of periodicity (e.g., calendar months have different number of timesteps).

https://github.com/CliMA/ClimaAtmos.jl/blob/719ee49f23492773bdf9bf4ad224100445d0d485/src/callbacks/callback_helpers.jl#L58-L75